modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_8_0/UpgradeDDMStructure.java

```diff
@@ -0,0 +1,399 @@
+import com.liferay.dynamic.data.mapping.io.DDMFormDeserializerDeserializeRequest;
+import com.liferay.dynamic.data.mapping.io.DDMFormDeserializerDeserializeResponse;
+import com.liferay.dynamic.data.mapping.io.DDMFormLayoutDeserializer;
+import com.liferay.dynamic.data.mapping.io.DDMFormLayoutDeserializerDeserializeRequest;
+import com.liferay.dynamic.data.mapping.io.DDMFormLayoutDeserializerDeserializeResponse;
+import com.liferay.dynamic.data.mapping.io.DDMFormLayoutSerializer;
+import com.liferay.dynamic.data.mapping.io.DDMFormLayoutSerializerSerializeRequest;
+import com.liferay.dynamic.data.mapping.io.DDMFormLayoutSerializerSerializeResponse;
+import com.liferay.dynamic.data.mapping.io.DDMFormSerializer;
+import com.liferay.dynamic.data.mapping.io.DDMFormSerializerSerializeRequest;
+import com.liferay.dynamic.data.mapping.io.DDMFormSerializerSerializeResponse;
+import com.liferay.dynamic.data.mapping.model.DDMForm;
+import com.liferay.dynamic.data.mapping.model.DDMFormField;
+import com.liferay.dynamic.data.mapping.model.DDMFormInstance;
+import com.liferay.dynamic.data.mapping.model.DDMFormLayout;
+import com.liferay.dynamic.data.mapping.model.DDMFormLayoutColumn;
+import com.liferay.dynamic.data.mapping.model.DDMFormLayoutPage;
+import com.liferay.dynamic.data.mapping.model.DDMFormLayoutRow;
+import com.liferay.dynamic.data.mapping.model.LocalizedValue;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.Tuple;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Marcos Martins
+ */
+public class UpgradeDDMStructure extends UpgradeProcess {
+
+	public UpgradeDDMStructure(
+		DDMFormDeserializer ddmFormDeserializer,
+		DDMFormLayoutDeserializer ddmFormLayoutDeserializer,
+		DDMFormLayoutSerializer ddmFormLayoutSerializer,
+		DDMFormSerializer ddmFormSerializer, JSONFactory jsonFactory) {
+
+		_ddmFormDeserializer = ddmFormDeserializer;
+		_ddmFormLayoutDeserializer = ddmFormLayoutDeserializer;
+		_ddmFormLayoutSerializer = ddmFormLayoutSerializer;
+		_ddmFormSerializer = ddmFormSerializer;
+		_jsonFactory = jsonFactory;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_upgradeDDMStructureLayout();
+		_upgradeDDMStructureVersion();
+		_upgradeDDMStructure();
+	}
+
+	private String _generateFieldSetName(DDMForm ddmForm) {
+		Map<String, DDMFormField> ddmFormFieldsMap =
+			ddmForm.getDDMFormFieldsMap(false);
+
+		String name = "FieldsGroup";
```

Comment:
@marcosapmf, why the method is named `_generateFieldSetName` and here the name is `FieldsGroup`? Can we rename to `ddmFormFieldName`?

In addition, @marcosapmf, will this name be the same once https://issues.liferay.com/browse/LPS-114619 get merged? When this story get merged then the Forms fields will be named according to the following pattern `Field12345678`.

Reply:
@natocesarrego, a fieldSet created by the UI is named as FieldGroup. We tried following this behavior as well. In this case, should we rename `_generateFieldSetName` or change `ddmFormFieldName's` value?